mainnet-contracts/src/CarrotVesting.sol

```diff
@@ -52,84 +58,55 @@ contract CarrotVesting is Ownable2Step {
 
-    // SLOT 1 - Along with the `_pendingOwner` address (160 + 48 + 8 = 216)
-    uint48 public pufferRecoveryStartTimestamp;
-    PufferRecoveryStatus public pufferRecoveryStatus;
-
-    // SLOT 2 - (48 + 32 + 32 + 128 = 240)
-    uint48 public startTimestamp = type(uint48).max; // Default value is max uint48 instead of 0 to avoid 2 checks in deposit function
-    uint32 public duration;
-    uint32 public steps;
-    uint128 public totalDepositedAmount;
-
-    // SLOT 3
-    mapping(address user => Vesting vestingInfo) public vestings;
-
-    constructor(address carrot, address puffer, address initialOwner) Ownable(initialOwner) {
+    constructor(address carrot, address puffer) {
         require(carrot != address(0), InvalidAddress());
         require(puffer != address(0), InvalidAddress());
         CARROT = IERC20(carrot);
         PUFFER = IERC20(puffer);
+        _disableInitializers();
     }
 
     /**
      * @notice Initializes the contract
      * @dev This function can only be called once by the owner
-     * @param _startTimestamp The timestamp when the vesting starts
-     * @param _duration The duration of the vesting (seconds since the user deposits)
-     * @param _steps The number of steps in the vesting (Example: If the vesting is 6 months and the user can claim every month, steps = 6)
-     */
-    function initialize(uint48 _startTimestamp, uint32 _duration, uint32 _steps) external onlyOwner {
-        require(startTimestamp == type(uint48).max, AlreadyInitialized());
-        require(_startTimestamp >= block.timestamp, InvalidStartTimestamp());
-        require(_duration > 0, InvalidDuration());
-        require(_steps > 0, InvalidSteps());
-
-        startTimestamp = _startTimestamp;
-        duration = _duration;
-        steps = _steps;
+     * @param startTimestamp The timestamp when the vesting starts
+     * @param duration The duration of the vesting (seconds since the user deposits)
+     * @param steps The number of steps in the vesting (Example: If the vesting is 6 months and the user can claim every month, steps = 6)
+     * @param initialOwner The address of the owner of the contract
+     */
+    function initialize(uint48 startTimestamp, uint32 duration, uint32 steps, address initialOwner)
+        external
+        initializer
+    {
+        require(startTimestamp >= block.timestamp, InvalidStartTimestamp());
+        require(duration > 0, InvalidDuration());
+        require(steps > 0, InvalidSteps());
+        require(duration >= steps, InvalidDuration());
+        require(initialOwner != address(0), InvalidAddress());
+
+        __Ownable_init(initialOwner);
+        __Pausable_init();
+        __UUPSUpgradeable_init();
+        __Ownable2Step_init();
+
+        VestingStorage storage $ = _getCarrotVestingStorage();
+        $.startTimestamp = startTimestamp;
+        $.duration = duration;
+        $.steps = steps;
         PUFFER.safeTransferFrom(msg.sender, address(this), TOTAL_PUFFER_REWARDS);
```

Comment:
This would mean that the deployer needs to have PUFFER approved to this contract.

Maybe separate initialize, and `initializeVesting` or name it whatever.
By doing that, initialize can be called during the deployment, the ownership can be granted to the correct multisig, and then that multisig can do
approve + `initializeVesting`